### DISCUSSION TITLE
Integration with Adwaita Colors

### DISCUSSION THREAD

**Discussion by somepaulo** [somepaulo/MoreWaita#237] [Ideas]
With the release of [Adwaita Colors](https://github.com/dpejoh/Adwaita-colors/) and [Auto Adwaita Colors](https://github.com/celiopy/auto-adwaita-colors) we should come up with a way of integrating MoreWaita and Adwaita Colors seamlessly, so users don't have to go through the hassle of changing things manually, but still have the choice of using MoreWaita or not.

I imagine five use cases in sort of three layers:  
1a. Adwaita Colors – just adding the colored folders for those who are fine with default icons and probably have one favourite colour they'll set and forget.  
1b. MoreWaita for those who just want that. 
2a. Adwaita Colors + Auto Adwaita Colors
2b. Adwaita Colors + MoreWaita
3. All three

Discuss ))

**Comment by dpejoh** [comment]
It can be done with shell script, I've already completed half of it, and I think I'll finish it today

**Comment by somepaulo** [comment]
For Arch I think I can do an AUR package that would detect if MoreWaita is installed, and if yes then just install what's missing in MoreWaita.  
And then I can trying the same from the MoreWaita side if I can make the PKGBUILDs delete the conflicting files if Adwaita Colors is already installed. I'll research this.

**Comment by dpejoh** [comment]
Good, take your time!

**Comment by somepaulo** [comment]
Just realised that the `mimes` folder should be called `mimetypes` to match Adwaita. I'm going to change that with the next point release of MoreWaita, aligned with Gnome, on the 23rd. 9 days.

**Comment by somepaulo** [comment]
One more thought. Can you test whether anything breaks if you have `Inherits=MoreWaita,Adwaita,AdwaitaLegacy,hicolor` by default in your index.theme without having MoreWaita installed?  
In theory, Gnome should just go through themes in that order skipping what it can't find until it finds it down the line, so there shouldn't be any trouble.

Not that it matters much, but it would be one less thing to worry about for packagers. And maybe  a couple of lines less code in your script.

**Comment by dpejoh** [comment]
Keeping `mimes` doesn't break Adwaita-colors with MoreWaita

> Can you test whether anything breaks if you have Inherits=MoreWaita,Adwaita,AdwaitaLegacy,hicolor by default in your index.theme without having MoreWaita installed?

It works perfectly.

**Comment by somepaulo** [comment]
> Keeping `mimes` doesn't break Adwaita-colors with MoreWaita

It's not so much about breaking, it just makes it easier to check against Adwaita. Diffing the theme folders didn't seem to show anything until I saw the folder name difference and changed the name locally. Then I found that Adwaita had two new mimetype icons that I had in MoreWaita. I don't want to override them, so I removed them.

**Comment by somepaulo** [comment]
I see you don't have a licence set for Adwaita Colors. You should add one. MoreWaita uses GPL3 because that's what other themes seem to use more often.

**Reply by dpejoh** [reply]
done

**Comment by somepaulo** [comment]
I've deployed two AUR packages.  
One [release-based](https://aur.archlinux.org/packages/adwaita-colors-icon-theme):  
```sh
paru adwaita-colors-icon-theme
```

And one [git-based](https://aur.archlinux.org/packages/adwaita-colors-icon-theme-git) for keeping up with all git changes between releases:  
```sh
paru adwaita-colors-icon-theme-git
```

*Replace `paru` with `yay` or your favourite AUR helper.

These packages assume that including `MoreWaita` at the start of the `Inherits` line does no harm if `morewaita-icon-theme` or `morewaita-icon-theme-git` is not present, so it just modifies all the `index.theme` files and installs your themes like that.

I suggest we test it like this on AUR users for a while, including myself, and if there are no complaints you could include this in your upstream `Inherit` by default at some later point.

Also, I'm a bit confused about your instruction to delete files from `scalable/`. I diffed all of your colour themes against MoreWaita, and there are no conflicting files at all. So my AUR packages don't delete any files.

EDIT: I'm happy to make you a co-maintainer on AUR if you want to. Just need to know your AUR handle.

**Reply by dpejoh** [reply]
Thank you for your work on the AUR packages! I had considered this before but don’t have any experience with creating packages, so I really appreciate your effort. I plan to make packages for Debian, Fedora, and other distributions in the future, and I’ll be learning along the way to handle everything myself. With the help of another developer on the extension, the project will become even more polished.

For now, it wouldn’t add much benefit to the project for me to be a co-maintainer :) We can consider it in the future.

**Reply by somepaulo** [reply]
You're welcome. If at any moment you want to co-maintain or even take over the AUR packages, just let me know.

**Comment by somepaulo** [comment]
On a separate note, when people use Adwaita Colors without MoreWaita, they get the extra folders (like Bitwig, etc.) only in the coloured versions, but not in the default blue one, since you wisely leave the defaults to Adwaita. It's probably a non-issue though, because if somebody installs Adwaita Colors, they probably don't want to use the default blue anyway.

What do you think?

**Reply by dpejoh** [reply]
Yes, that's correct. But are you suggesting that it would be a good idea to have the extra themes also installed for Adwaita? Anyway, MoreWaita users already have them

**Reply by somepaulo** [reply]
No, not really. Just a thought. It probably won't even be a use case. You can have MoreWaita installed, but inactive, and still assign any of its extra folder icons to your folders.  
Similar to my use case, actually. I like the default blue, but now that I have Adwaita Colors installed, I can assign icons of different colours to some folders without activating any of the coloured themes. Profit!

**Reply by dpejoh** [reply]
The user now has plenty of choices! XD

**Comment by somepaulo** [comment]
Four more folders have landed through new PRs. You may want to ingest those. They are:  
Globe, World, Google Drive and OneDrive

**Reply by dpejoh** [reply]
Sure!

**Comment by dpejoh** [comment]
Hello,

MoreWaita now works perfectly with Adwaita-colors following the latest update!
